lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,63 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $destination Path (`container/object') of other object to symlink this object to
+     * @return null|\Guzzle\Http\Message\Response The response or null if $this is not empty
```

Comment:
Hey @markchalloner, could you please remove the `null` from the @return type and description? Also, could you please add the necessary [@throws](http://phpdoc.org/docs/latest/references/phpdoc/tags/throws.html) documentation for the exceptions thrown by this method?